lib/checksum.js

```diff
@@ -34,14 +47,16 @@ const utils = require('./utils');
  * @param {Object} options - options
  * @param {Number} options.imageSize - image size in bytes
  * @param {Number} options.chunkSize - chunk size
+ * @param {String[]} options.algorithms=['crc32'] - checksum hash algorithms
```

Comment:
The reasoning behind that came from https://github.com/resin-io/etcher/issues/643#issuecomment-241761064, and also from thinking that this will make it a lot easier to add / swap out hash streams in the future.
This way we can just plug & play checksum algorithms, and support all node core hash types (everything that `openssl list-message-digest-algorithms` outputs on the given platform) without any additional work.